resources/qml/PrinterSelector/MachineSelector.qml

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2018 Ultimaker B.V.
+// Cura is released under the terms of the LGPLv3 or higher.
+
+import QtQuick 2.7
+import QtQuick.Controls 2.3
+
+import UM 1.2 as UM
+import Cura 1.0 as Cura
+
+Cura.ExpandableComponent
+{
+    id: machineSelector
+
+    property bool isNetworkPrinter: Cura.MachineManager.activeMachineNetworkKey != ""
+    property bool isPrinterConnected: Cura.MachineManager.printerConnected
+    property var outputDevice: Cura.MachineManager.printerOutputDevices.length >= 1 ? Cura.MachineManager.printerOutputDevices[0] : null
+
+    popupPadding: UM.Theme.getSize("default_lining").width
+    popupAlignment: Cura.ExpandableComponent.PopupAlignment.AlignLeft
+    iconSource: expanded ? UM.Theme.getIcon("arrow_bottom") : UM.Theme.getIcon("arrow_left")
+
+    UM.I18nCatalog
+    {
+        id: catalog
+        name: "cura"
+    }
+
+    headerItem: Cura.IconLabel
+    {
+        text: isNetworkPrinter ? Cura.MachineManager.activeMachineNetworkGroupName : Cura.MachineManager.activeMachineName
+        source:
+        {
+            if (isNetworkPrinter)
+            {
+                if (machineSelector.outputDevice != null && machineSelector.outputDevice.clusterSize > 1)
+                {
+                    return UM.Theme.getIcon("printer_group")
+                }
+                return UM.Theme.getIcon("printer_single")
+            }
+            return ""
+        }
+        font: UM.Theme.getFont("medium")
+        color: UM.Theme.getColor("text")
+        iconSize: UM.Theme.getSize("machine_selector_icon").width
+
+        UM.RecolorImage
+        {
+            id: icon
+
+            anchors.bottom: parent.bottom
+            x: UM.Theme.getSize("thick_margin").width
```

Comment:
Why use x and not an anchor?

Reply:
Done in 5b940b524204ca56b5fafe05bad28d5e477b8ea7